src/PMGDQueryHandler.cc

```diff
@@ -169,6 +169,12 @@ int PMGDQueryHandler::process_query(const PMGDCmd *cmd,
         throw e;
     }
 
+    if (retval < 0) {
```

Comment:
This is not needed here. It can go in the error_cleanup function since we already call it on error return.

Reply:
Won't work. error_cleanup can be called after unlocking _dblock, std::mutex has undefined behavior if you unlock a lock twice. 
The safe place to unlock is the same function we lock ( process_query(2) ).